pkg/resources/resource_cluster.go

```diff
@@ -87,29 +94,37 @@ func Cluster() *schema.Resource {
 		DeleteContext: clusterDelete,
 
 		Importer: &schema.ResourceImporter{
-			StateContext: schema.ImportStatePassthroughContext,
+			StateContext: clusterImport,
 		},
 
 		Schema: clusterSchema,
 	}
 }
 
 func clusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	i := d.Id()
-
+	fullId := d.Id()
 	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	s, err := materialize.ScanCluster(metaDb, utils.ExtractId(i))
+
+	idType := utils.ExtractIdType(fullId)
+	value := utils.ExtractId(fullId)
+	useNameAsId := d.Get("identify_by_name").(bool)
+
+	s, err := materialize.ScanCluster(metaDb, value, idType == "name")
 	if err == sql.ErrNoRows {
 		d.SetId("")
 		return nil
 	} else if err != nil {
 		return diag.FromErr(err)
 	}
 
-	d.SetId(utils.TransformIdWithRegion(string(region), i))
+	if useNameAsId {
+		d.SetId(utils.TransformIdWithTypeAndRegion(string(region), "name", s.ClusterName.String))
+	} else {
+		d.SetId(utils.TransformIdWithTypeAndRegion(string(region), "id", s.ClusterId.String))
+	}
```

Comment:
There should be no need for a state migration, as we will adjust the ID format accordingly directly in the read func.

Reply:
Ah ok, nice!